v1/translate.yaml

```diff
@@ -0,0 +1,254 @@
+info:
+  version: '1.0.0'
+  title: Content Translation API
+  description: Content Translation API
+  termsofservice: https://github.com/wikimedia/restbase
+  license:
+    name: Apache licence, v2
+    url: https://www.apache.org/licenses/LICENSE-2.0
+paths:
+  /machine/{from}/{to}{/provider}:
+    post:
+      tags: ['Translate']
+      summary: Machine-translate content
+      description: |
+        Fetches the machine translation for the posted content from the source
+        to the destination language.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      consumes:
+        - application/x-www-form-urlencoded
+      produces:
+        - application/json
+      parameters:
+        - name: from
+          in: path
+          description: The source language code
+          type: string
+          required: true
+        - name: to
+          in: path
+          description: The target language code
+          type: string
+          required: true
+        - name: provider
+          in: path
+          description: The machine translation provider id
+          type: string
+          required: false
+          enum:
+           - Apertium
```

Comment:
-1 but not -2 from me. I don't think there will be many errors, the users should figure out the list of available engines via the listing APIs, not typing them manually. Hence they are not fixed. In fact the enum is wrong and does not list all possible values (which again depend on the language pair). Apertium was used as an example in the old API because it is the only service that does not require an authorization header (which is missing from these docs btw).

See also https://gerrit.wikimedia.org/r/#/c/346991/8/modules/mw.cx.MachineTranslationManager.js for an example where this pattern is taken as the convention.